mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -766,6 +767,28 @@ class SparseVector(
       maxIdx
     }
   }
+
+  /**
+   * Create a slice of this vector based on the given indices.
+   * @param selectedIndices Sorted list of indices into the vector. This does NOT do bound checking.
+   * @return  New SparseVector with values in the order specified by the given indices.
+   *
+   * NOTE: The API needs to be discussed before making this public.
+   */
+  private[spark] def slice(selectedIndices: Array[Int]): SparseVector = {
+    var currentIdx = 0
+    val (sliceInds, sliceVals) = selectedIndices.flatMap { origIdx =>
```

Comment:
Also leave a TODO to optimize the implementation. If we assume input indices are ordered (or sort it), then this should be a merge intersection.